### PR TITLE
Feature/roe 1406 and 1430  move create transaction and oe to presenter page

### PR DESCRIPTION
**Pull request by mouhajer-ch**
### JIRA link

Resolves [ROE-1406](https://companieshouse.atlassian.net/browse/ROE-1406) and [ROE-1430](https://companieshouse.atlassian.net/browse/ROE-1430)

### Change description

Move create `Transaction` and `Overseas Entities` to the `presenter` page

### Work checklist

- [x] Tests added where applicable
- [ ] UI changes meet accessibility criteria

### CODE REVIEW

File: src/controllers/check.your.answers.controller.ts
Comment (markpit) [inline src/controllers/check.your.answers.controller.ts:66]: we'll need to do a PUT to the OE API if feature flag is active as the only data that will have been sent so far is the presenter data (once PUT endpoint is done)

Comment (mouhajer-ch) [reply src/controllers/check.your.answers.controller.ts:66]: Yep, it is part of [ROE-1407](https://companieshouse.atlassian.net/browse/ROE-1407)

File: src/controllers/check.your.answers.controller.ts
Comment (mwestacott) [inline src/controllers/check.your.answers.controller.ts:64]: Variable name here not quite right - missing an 's' at end of 'oversea'.

But also, looks a bit odd - a true/false value is being assigned to the ID?

File: src/controllers/check.your.answers.controller.ts
Comment (mwestacott) [inline src/controllers/check.your.answers.controller.ts:60]: I'm also not sure about the logic here. Might be easier to have a call to understand what's going on.

File: src/controllers/check.your.answers.controller.ts
Comment (mouhajer-ch) [reply src/controllers/check.your.answers.controller.ts:64]: It is a conditional (ternary) operator

File: src/controllers/check.your.answers.controller.ts
Comment (mwestacott) [inline src/controllers/check.your.answers.controller.ts:60]: Here and below, I think it would be easier to read code if you removed the ! in front of the `isSaveAndResumeFeatureActive` variable and swapped around the following logic.

File: src/controllers/check.your.answers.controller.ts
Comment (mwestacott) [reply src/controllers/check.your.answers.controller.ts:64]: Yes, misread this. Variable name still needs changing and see comment above about not using ! here.

File: src/controllers/check.your.answers.controller.ts
Comment (mouhajer-ch) [reply src/controllers/check.your.answers.controller.ts:60]: ok, sure

File: src/controllers/presenter.controller.ts
Comment (mwestacott) [inline src/controllers/presenter.controller.ts:45]: Could make this a TODO comment with reference to story JIRA id.

**Review by mwestacott — DISMISSED** [review]

File: src/routes/index.ts
Comment (mwestacott) [inline src/routes/index.ts:33]: So yes, as discussed this needs to remain now it's been clarified.

**Review by mwestacott — COMMENTED** [review]
Need to sort out the `routes/index.ts` file.

File: src/routes/index.ts
Comment (mouhajer-ch) [reply src/routes/index.ts:33]: > 

Thanks @mwestacott, I will fix it first think in the morning. Do you know how to dismiss your approval?

**Comment by mouhajer-ch** [comment]
> Need to sort out the `routes/index.ts` file.

I have dismissed your approval until fix is done

File: src/routes/index.ts
Comment (mouhajer-ch) [reply src/routes/index.ts]: Done

**Comment by markpit** [comment]
looks ok to me, should wait for @mwestacott to confirm his comments have been resolved

**Pull request merged**
